src/Symfony/Component/Messenger/Handler/Locator/AbstractHandlerLocator.php

```diff
@@ -21,21 +19,19 @@
  */
 abstract class AbstractHandlerLocator implements HandlerLocatorInterface
 {
-    public function getHandler(Envelope $envelope): ?callable
+    public function getHandler(string $messageName): ?callable
     {
-        $class = \get_class($envelope->getMessage());
-
-        if ($handler = $this->getHandlerByName($class)) {
+        if ($handler = $this->getHandlerByName($messageName)) {
             return $handler;
         }
 
-        foreach (class_parents($class) as $name) {
+        foreach (class_parents($messageName) as $name) {
```

Comment:
can we really assume the messageName is a valid class now? https://3v4l.org/OM16J